geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HlenDUnitTest.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.dunit.rules.RedisClusterStartupRule;
+import org.apache.geode.test.junit.rules.ExecutorServiceRule;
+
+public class HlenDUnitTest {
+
+  @ClassRule
+  public static RedisClusterStartupRule cluster = new RedisClusterStartupRule();
+
+  @ClassRule
+  public static ExecutorServiceRule executor = new ExecutorServiceRule();
+
+  private static final int HASH_SIZE = 5000;
+  private static final int NUM_ITERATIONS = 50000;
+  private static MemberVM locator;
+  private static MemberVM server1;
+  private static MemberVM server2;
+  private static int[] redisPorts;
+  private static RedisCommands<String, String> lettuce;
+  private static StatefulRedisConnection<String, String> connection;
+  private static ClientResources resources;
+
+  @BeforeClass
+  public static void classSetup() {
+    redisPorts = AvailablePortHelper.getRandomAvailableTCPPorts(3);
+
+    String redisPort1 = String.valueOf(redisPorts[0]);
+    String redisPort2 = String.valueOf(redisPorts[1]);
+
+    locator = cluster.startLocatorVM(0);
+
+    server1 = startRedisVM(1, redisPorts[0]);
+    server2 = startRedisVM(2, redisPorts[1]);
+
+    DUnitSocketAddressResolver dnsResolver =
+        new DUnitSocketAddressResolver(new String[] {redisPort2, redisPort1});
+
+    resources = ClientResources.builder().socketAddressResolver(dnsResolver).build();
+
+    RedisClient redisClient = RedisClient.create(resources, "redis://localhost");
+    redisClient.setOptions(ClientOptions.builder().autoReconnect(true).build());
+
+    connection = redisClient.connect();
+    lettuce = connection.sync();
+  }
+
+  private static MemberVM startRedisVM(int vmID, int redisPort) {
+    int locatorPort = locator.getPort();
+
+    return cluster.startRedisVM(vmID, (x) -> x
+        .withConnectionToLocator(locatorPort)
+        .withProperty(REDIS_PORT, "" + redisPort));
+  }
+
+  @Before
+  public void testSetup() {
+    lettuce.flushall();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    resources.shutdown().get();
+    connection.close();
+
+    server1.stop();
+    server2.stop();
+  }
+
+  @Test
+  public void testConcurrentHLens_returnExpectedLength() {
+    AtomicLong client1Len = new AtomicLong();
+    AtomicLong client2Len = new AtomicLong();
+
+    String key = "HLEN";
+
+    Map<String, String> setUpData = makeInitialHashMap(HASH_SIZE, "field", "value");
+
+    lettuce.hset(key, setUpData);
+
+    new ConcurrentLoopingThreads(NUM_ITERATIONS,
+        i -> {
+          long len = lettuce.hlen(key);
+          client1Len.addAndGet(len);
+        },
+        i -> {
+          long len = lettuce.hlen(key);
+          client2Len.addAndGet(len);
+        })
+            .run();
+
+    assertThat(client1Len.get() + client2Len.get()).isEqualTo(NUM_ITERATIONS * HASH_SIZE * 2);
+  }
+
+  @Test
+  public void testConcurrentHLen_whileAddingFields() {
```

Comment:
I was just thinking, is there value to adding a concurrent HLEN test while deleting fields as well?

Reply:
good idea! added in recent commits